pkg/ring/model.go

```diff
@@ -45,18 +45,17 @@ func (d *Desc) AddIngester(id, addr string, tokens []uint32, state IngesterState
 		Addr:      addr,
 		Timestamp: time.Now().Unix(),
 		State:     state,
+		Tokens:    tokens,
 	}
 
-	if normaliseTokens {
-		ingester.Tokens = tokens
-	} else {
-		for _, token := range tokens {
-			d.Tokens = append(d.Tokens, TokenDesc{
-				Token:    token,
-				Ingester: id,
-			})
+	// delete any reference from d.Tokens (can happen if denormalized tokens for this ingester
```

Comment:
I would rephrase the comment for clarity, explaining we do `delete any denormalized token for this ingester, given all ingester tokens are going to be stored normalized. It may happen to find denormalized tokens when joining the ring (ie. the previous instance of the same ingester was running with denormalized tokens).`

Reply:
I've updated comment, and also added test for this. It turns out that my previous test was checking different scenario (takeover of denormalised tokens by different ingester), and commenting out this code didn't make it fail.